sound/soc/sof/intel/hda-stream.c

```diff
@@ -486,7 +484,7 @@ irqreturn_t hda_dsp_stream_threaded_handler(int irq, void *context)
 				continue;
 
 			/* Inform ALSA only in case not do that with IPC */
-			if (sof_hda->no_ipc_position)
+			if (s->no_period_wakeup)
```

Comment:
why are you calling period elapse here when no period wakeup?

Reply:
I didn't get the usage of this code, i changed this aligned with previous code. In my code this is in IRQ handler which is not used in nowakeup mode.